lib/babe/babe.go

```diff
@@ -265,6 +265,24 @@ func (b *Service) SetAuthorities(a []*types.BABEAuthorityData) {
 	b.authorityData = a
 }
 
+// SetBlockProducerAuthorities sets the current Block Producer Authorities and sets Authority index
+func (b *Service) SetBlockProducerAuthorities(data []*types.BABEAuthorityData) error {
```

Comment:
i think you can rename this to `SetAuthorities` and delete the existing `SetAuthorities` function, unless that causes a lot of stuff to break?